inspirehep/modules/records/tasks.py

```diff
@@ -0,0 +1,153 @@
+
+    with db.session.begin_nested():
+        for rec_uuid in record_uuids:
+            rec_db = Record.get_record(rec_uuid)
+            find_and_replace_old_link_with_new_link_in_record(rec_db, new_record_ref, old_record_ref)
+            rec_db.commit()
+    db.session.commit()
+
+
+def get_recid_from_url(url):
+    try:
+        res = int(url.split('/')[-1])
+    except ValueError:
+        res = None
+    return res
+
+
+def find_ref_fields_in_schema(schema_name):
+    def construct_str(string):
+        result_list = []
+        for splited_path in string.split('.'):
+            if '_records' in splited_path:
+                result_list.append(splited_path[:-len('_records')] + '_recid')
+                break
+            elif '_record' in splited_path:
+                result_list.append(splited_path[:-len('_record')] + '_recid')
+                break
+            elif 'records' in splited_path:
+                result_list.append(splited_path[:-len('records')] + 'recid')
+                break
+            elif 'record' in splited_path:
+                result_list.append(splited_path[:-len('record')] + 'recid')
+                break
+
+            result_list.append(splited_path)
+        return '.'.join(result_list)[1:]
+
+    schema = api.load_schema(schema_name)
+    all_fields_dotted = dotter(schema.get('properties'), '', [])
+    fields = []
+    for field_dotted in all_fields_dotted:
+        if 'record' in field_dotted:
+            contructed_field = construct_str(field_dotted)
+            fields.append(contructed_field)
+
+    return fields
+
+
+def get_ref_fields(ignore_refs_in_fields, schemas_names):
+    if schemas_names is None:
+        schemas_names = ['hep', 'authors', 'conferences',
+                         'experiments', 'institutions',
+                         'jobs', 'journals']
+    fields_to_check_links = []
+
+    for schema_name in schemas_names:
+        fields_to_check_links += find_ref_fields_in_schema(schema_name)
+
+    set_fields_to_check_links = set(fields_to_check_links)
+    set_bad_fields = set(ignore_refs_in_fields)
+
+    return list(set_fields_to_check_links - set_bad_fields)
+
+
+def es_search_pointing_ids_from_ref_in_terms(fields_recid, old_control_number):
+    body = {
+        "query": {
+            "bool": {
+                "should": [
+                    {
+                        "term": {
+                            _key: {"value": old_control_number}}
+                    }
+                    for _key in fields_recid]
+            }
+        },
+        "fields": [
+            "_id"
+        ]
+    }
+
+    scroll = scan(es, query=body, index="records-*")
+
+    for response in scroll:
+        rec_uuid = response['_id']
+
+        yield rec_uuid
+
+
+def find_and_replace_old_link_with_new_link_in_record(record, new_link, old_link):
+    bad_fields = ['deleted_recid', 'new_recid']
```

Comment:
But why is it a blacklist, anyway? This means that if some field with a `$ref` is added you need to remember to update this field if needed, otherwise data loss happens. If it's a whitelist then at worst you lose an update, which is going to be done by the background task anyway.